ENIGMAsystem/SHELL/Platforms/xlib/XLIBmain.cpp

```diff
@@ -194,14 +196,36 @@ void handleInput() {
 
 }  // namespace enigma
 
+namespace {
+
+int display_get_helper(unsigned i) {
+  int result = 0;
+  string output = enigma_user::execute_shell_for_output("xrandr | awk '/primary/ { print $4 }'");
+  output = enigma_user::string_replace_all(output, "x", "|");
+  output = enigma_user::string_replace_all(output, "+", "|");
+  var split_output = enigma_user::string_split(output, "|");
+  result = std::stoi(split_output[i], nullptr, 10);
```

Comment:
if you want to prevent the ci from crashing you need to check if i is < split_output.size() before calling this. However, this whole approach seems like a hack and there should be a way to do it in c++ without calling bash

Reply:
There is a way to do it in C++ but it requires a large dependency, which we already use in the xrandr extension for changing screen resolution, i would like that dependency to remain only needed with that extension and not by the whole engine; that was initially Roberts idea to make it an extension. Unfortunately what he didn't know is that package is needed not just for setting display size of a given monitor; it's also needed for getting the size.

Reply:
you can do it with pure xlib. theres no extra deps needed

Reply:
libxrandr is a part of the x11 standard. it wouldn't exist if it could be also done by some other x11 library. Stuff like XDisplayWidth and such can only get the display size of all the monitors combined. I could be wrong about this, but that's just the impression I got from researching this topic for linterally years. I would strongly prefer to do it in pure xlib if I knew how.

Reply:
randr just uses xlib internally

Reply:
https://tronche.com/gui/x/xlib/display/display-macros.html#ScreenCount if you can get the count im sure you can get the size of indivdual screens

Reply:
I did actually try that already. I have two monitors connected. using 0 for the screen count returns returns the size of the two monitors combined. Any higher than that i get some nonsense negative number for both the width and height. (Yes, I tried both values of 1 and 2).

Reply:
I'm pretty sure it uses libxrandr and internally, and i even viewed it's source code which doesn't use that. https://gitlab.freedesktop.org/xorg/lib/libxrandr/tree/master/src Although I just had a quick glance. I could've missed it.

Reply:
This file is probably the most relevent: https://gitlab.freedesktop.org/xorg/lib/libxrandr/blob/master/src/Xrandr.c

Reply:
yes xrandr uses xlib. xlib doesnt use xrandr
https://gitlab.freedesktop.org/xorg/lib/libxrandr/blob/master/src/Xrandr.c#L30

Reply:
It's not using it for actually getting the resolution. The only thing it even appears to be using from that header is the Display * data type.

Reply:
and Display is a struct with things like resolution

Reply:
I really don't know what I'm talking about, I'm just saying what I can think based on a limited point of view.

Reply:
What I'm saying is, the point of this file is to utilize the display struct to get such info. I'm merely agreeing with you. But the problem is I don't think the X.org devs went through the trouble to reinvent the wheel in two different places.

Reply:
Display* d = XOpenDisplay(NULL);
Screen*  s = DefaultScreenOfDisplay(d);
return s->width;

it should be something like that

Reply:
I've also tried that exact thing http://surfingtroves.blogspot.com/2011/01/how-to-get-screen-resolution-in-linux-c.html

That link is where i got the idea before you mentioned it. Again, it can only return the display size of all screens combined.

Reply:
In any case, this does actually work on my machine. I don't know why it isn't in the ci when travis is also using the same version of Ubuntu my KDE neon is based on.

Reply:
its crashing because the randr cmd you ran returned something different on the ci, your code expects a certain output but there'sx no guarantee randr is going to print the same thing on all pcs

Reply:
your code makes a lot of assumptions and is prone to fail

Reply:
I see. Well I'll keep this pr open until I find some other means that works more reliably.

Reply:
@fundies I made the code parse the output a little more reliably. It still crashed the ci. It's probably because the ci doesn't have xrandr installed for some odd reason. So what I did is check for failure to prevent the crash and on failure fall back on the former behavior which returns the size of all monitors combined.

Reply:
My XRandR extension was merged, hugar tested it and said it works.

Reply:
@RobertBColton I wasn't arguing whether it works. I was stating it is being used as an extension and not a mandatory part of the engine. window_center() if a part of platforms_mandatorh.h which uses display_get_width() and display_get_height(). We can't put the display getting functions in that extension for this reason, which is why I am using xrandr CLI for the display getter, and not the library like the setter functions that your extension offers.

Reply:
I'm trying to find info about `<X11/extensions/randr.h>`, but it's not coming up easily. It's quite possible everything we need from librandr is already on host systems.